src/request_builder.rs

```diff
@@ -0,0 +1,379 @@
+        self.initial_state = state;
+        self
+    }
+
+    /// Set the vec of `UserId`s.
+    pub fn invite(&mut self, invite: Vec<UserId>) -> &mut Self {
+        self.invite = invite;
+        self
+    }
+
+    /// Set the vec of `Invite3pid`s.
+    pub fn invite_3pid(&mut self, invite: Vec<Invite3pid>) -> &mut Self {
+        self.invite_3pid = invite;
+        self
+    }
+
+    /// Set the vec of `Invite3pid`s.
+    pub fn is_direct(&mut self, direct: bool) -> &mut Self {
+        self.is_direct = Some(direct);
+        self
+    }
+
+    /// Set the room name. A `m.room.name` event will be sent to the room.
+    pub fn name<S: Into<String>>(&mut self, name: S) -> &mut Self {
+        self.name = Some(name.into());
+        self
+    }
+
+    /// Set the room's power levels.
+    pub fn power_level_override(&mut self, power: PowerLevelsEventContent) -> &mut Self {
+        self.power_level_content_override = Some(power);
+        self
+    }
+
+    /// Convenience for setting various default state events based on a preset.
+    pub fn preset(&mut self, preset: RoomPreset) -> &mut Self {
+        self.preset = Some(preset);
+        self
+    }
+
+    ///  The local part of a room alias.
+    pub fn room_alias_name<S: Into<String>>(&mut self, alias: S) -> &mut Self {
+        self.room_alias_name = Some(alias.into());
+        self
+    }
+
+    /// Room version, defaults to homeserver's version if left unspecified.
+    pub fn room_version<S: Into<String>>(&mut self, version: S) -> &mut Self {
+        self.room_version = Some(version.into());
+        self
+    }
+
+    /// If included, a `m.room.topic` event will be sent to the room.
+    pub fn topic<S: Into<String>>(&mut self, topic: S) -> &mut Self {
+        self.topic = Some(topic.into());
+        self
+    }
+
+    /// A public visibility indicates that the room will be shown in the published
+    /// room list. A private visibility will hide the room from the published room list.
+    /// Rooms default to private visibility if this key is not included.
+    pub fn visibility(&mut self, vis: Visibility) -> &mut Self {
+        self.visibility = Some(vis);
+        self
+    }
+}
+
+impl Into<create_room::Request> for RoomBuilder {
+    fn into(self) -> create_room::Request {
+        create_room::Request {
+            creation_content: self.creation_content,
+            initial_state: self.initial_state,
+            invite: self.invite,
+            invite_3pid: self.invite_3pid,
+            is_direct: self.is_direct,
+            name: self.name,
+            power_level_content_override: self.power_level_content_override,
+            preset: self.preset,
+            room_alias_name: self.room_alias_name,
+            room_version: self.room_version,
+            topic: self.topic,
+            visibility: self.visibility,
+        }
+    }
+}
+
+/// Create a builder for making get_message_event requests.
+#[derive(Clone, Default)]
+pub struct RoomMessageBuilder {
```

Comment:
The name of this struct makes me think that this builds a room message that will be sent out, possibly from a markdown string or something like that.

Perhaps a better name would be `MessagesRequestBuilder`, `RoomMessagesRequestBuilder`, or `RoomMessagesRequests`.